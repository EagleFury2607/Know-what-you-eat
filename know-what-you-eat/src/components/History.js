import React, { Component } from "react";
import Axios from "axios";
import NavbarComponent from "./NavbarComponent";
import "./css/history.css";
class History extends Component {
  constructor(props) {
    super(props);
    this.state = {
      loading: false,
      data: [],
    };
    this.fetchHistory = this.fetchHistory.bind(this);
  }
  componentDidMount() {
    // console.log("Component is mounted !");
    this.fetchHistory();
  }
  fetchHistory = () => {
    const jwt = localStorage.getItem("jwt");
    if (!jwt) {
      this.props.history.push("/login");
    }
    Axios.get("/getHistory", {
      headers: {
        Authorization: `Bearer ${jwt}`,
      },
      option: {
        "Access-Control-Allow-Origin": "*",
      },
    })
      .then((result) => {
        this.setState({
          data: result.data[0][0].history,
        });
        // console.log(this.state.data);
      })
      .catch((err) => {
        console.log("ERR", err);
      });
  };
  render() {
    var printOut = "";
    if (this.state.data !== []) {
      console.log(this.state.data);
      printOut = this.state.data.map(function (data, id) {
        return (
          <div>
            <li key={id}>
              Date: {data.date}
              <br></br>
              Front Fruit: {data.frontFruitName}
              <br></br>
              Side Fruit: {data.sideFruitName}
              <br></br>
              Time Taken: {data.time}
              <br></br>
              Calorie: {data.calorie}
            </li>
          </div>
        );
      });
    }
    return (
      <div>
        <NavbarComponent userLogginStatus={"Logout"} />
        <h1>HISTORY</h1>
        <div
          id="showData"
          style={{
            display: "flex",
            alignItems: "center",
            flexDirection: "column",
          }}
        >
          <div className="list-type1">
            <ol>{printOut}</ol>
          </div>
        </div>
      </div>
    );
  }
}

export default History;
