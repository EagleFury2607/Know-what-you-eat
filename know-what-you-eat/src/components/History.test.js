import History from "./History";
import React, { Component } from "react";
import { shallow, EnzymeAdapter } from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
// import {jest} from '@jest/globals'

configure({ adapter: new Adapter() });
// import axios from "axios";
// import MockAdapter from "axios-mock-adapter";

// // at the time of testing , only one object was there
// // User : rohitsa4@gmail.com is being tested here
global.localStorage = {
  getItem: () => "3d8cc8bb-0579-412e-9005-0e213af876c0",
};
describe("Unit tests for History",()=>{
	test("History Component Loads Correctly", () => {
		const dashboard = shallow(<History />);
		expect(dashboard).toMatchSnapshot();
	});
	test("History Text displayed", () => {
		const dashboard = shallow(<History />);
		expect(dashboard.find("h1").text()).toBe("HISTORY");
	});
	test("List format is correct", () => {
		const dashboard = shallow(<History />);
		expect(dashboard.find("div.showData").exists);
	});
});
