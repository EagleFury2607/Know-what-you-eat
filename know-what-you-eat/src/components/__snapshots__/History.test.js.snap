exports[`Unit tests for History History Component Loads Correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <History />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withRouter(NavbarComponent)
          userLogginStatus="Logout" />,
        <h1>
          HISTORY
</h1>,
        <div
          id="showData"
          style={
                    Object {
                              "alignItems": "center",
                              "display": "flex",
                              "flexDirection": "column",
                            }
          }>
          <div
                    className="list-type1">
                    <ol>
                              
                    </ol>
          </div>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "userLogginStatus": "Logout",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "HISTORY",
        },
        "ref": null,
        "rendered": "HISTORY",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="list-type1">
            <ol>
                        
            </ol>
</div>,
          "id": "showData",
          "style": Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ol>
              
</ol>,
            "className": "list-type1",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
            },
            "ref": null,
            "rendered": Array [],
            "type": "ol",
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter(NavbarComponent)
            userLogginStatus="Logout" />,
          <h1>
            HISTORY
</h1>,
          <div
            id="showData"
            style={
                        Object {
                                    "alignItems": "center",
                                    "display": "flex",
                                    "flexDirection": "column",
                                  }
            }>
            <div
                        className="list-type1">
                        <ol>
                                    
                        </ol>
            </div>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "userLogginStatus": "Logout",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "HISTORY",
          },
          "ref": null,
          "rendered": "HISTORY",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="list-type1">
              <ol>
                            
              </ol>
</div>,
            "id": "showData",
            "style": Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "column",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ol>
                
</ol>,
              "className": "list-type1",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
              },
              "ref": null,
              "rendered": Array [],
              "type": "ol",
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
